addons/portal/static/src/js/portal_chatter.js

```diff
@@ -254,6 +257,26 @@ var PortalChatter = publicWidget.Widget.extend({
         var page = $(ev.currentTarget).data('page');
         this._changeCurrentPage(page);
     },
+
+    /**
+     * @private
+     * @returns {Promise}
+     */
+    _onClickUpdateIsInternal: function (ev) {
+        ev.preventDefault();
+
+        var $elem = $(ev.currentTarget);
+        return this._rpc({
+            route: '/mail/update_is_internal',
+            params: {
+                message_id: $elem.data('message-id'),
+                is_internal: $elem.data('is-internal'),
+            },
+        }).then(function (result) {
+            $elem.toggleClass('o_portal_message_internal_off o_portal_message_internal_on');
```

Comment:
The button works visually because this is a toggle class, but in current state it doesn't work (see other comments).

To avoid this kind of issues, I would avoid toggle here too, and use `addClass` and `removeClass` depending on the final value.

Reply:
Copy pasted from website stuff. I prefer keeping the old code if you plan to rewrite everything in owl anyway.

Reply:
You still need to change the `is-internal` data on the DOM here depending on the result of the RPC, otherwise the toggle button will not actually toggle, but always set the opposite state of the one that was existing when the page was first rendered.
Since this condition is necessary, also properly adding/removing the class ass suggested becomes a trivial change.

Reply:
Ok then ! You're the boss boss.